lib/src/source/hosted.dart

```diff
@@ -354,11 +363,13 @@ class BoundHostedSource extends CachedSource {
 
   /// Downloads package [package] at [version] from [server], and unpacks it
   /// into [destPath].
-  Future _download(
-      String server, String package, Version version, String destPath) async {
-    var url = Uri.parse('$server/packages/$package/versions/$version.tar.gz');
+  Future _download(PackageId id, String destPath) async {
+    final versions = await _scheduler.schedule(id.toRef());
```

Comment:
Won't this introduce an extra round-trip if installing from a `pubspec.lock` (I'm not sure, just guessing).

This might also be acceptable.

Reply:
Yes I think it would - not sure how much a problem this is, all the packages are AFAIK downloaded in parallel, so it will not take the time of an extra round-trip per package.

On the positive side, it gives us freedom to move things around by changing the archive_url.

If we get around to cache the response from api/package/versions on disk, the issue will diminish even further.